### PR TITLE
fix: View mode dimension observer

### LINKED ISSUE #28556: [Bug]: Auto-Layout - Height of Text Widget Not Fitting MultiLine Text Set Programmatically in Text Widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The user has enabled auto-layout and updates the text of a Text widget programmatically, at the click of a button that runs a JS function. If the height of the new content exceeds the initial height of the Text widget, the text is cropped and the height of the text widget does not increase to fit the new content. This issue occurs only in the deployed view of the app and works fine in edit mode.

### Steps To Reproduce

1. Add a **Text widget** and a **Button widget** to the canvas.
2. Create a JS function that sets the text of the Text widget programmatically, and run this function onClick of the button.
3. Ensure the new text set programmatically in the Text widget is multi-line and exceeds the height of the initial content.
4. Deploy the app and click the button to set the text programmatically (see screen recording and sample app).

### Public Sample App

https://app.appsmith.com/app/autolayout-text-height/page1-65424143e13a2c24b30e0014

### Environment

Production

### Issue video log

https://www.loom.com/share/b8579966032c4526af25f56c5662d372

### Version

v1.9.43

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- The dimension observer in auto layout was not triggering in the viewer due to the fact that the observed container did not adapt to the child contents.
- This worked in the editor because the document was not in the default `position:auto` flow.

- The fix was to change the `position` property to `auto` and make sure that the observed container has the height which `fit-content`.

Reference: https://www.joshwcomeau.com/css/understanding-layout-algorithms/

#### PR fixes following issue(s)
Fixes #28556

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [NA] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by shadabbuchh** [comment]
Height of Text Widget adjusts the multi line text set programmatically in text widget now. Also tested for the height of group widgets. 
https://www.loom.com/share/e66d61061eef4c1ab0e393b4ed52fa62

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoLayoutAutoHeight_spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+describe("Validating use cases for Auto Dimension", () => {
+  before(() => {
+    autoLayout.ConvertToAutoLayoutAndVerify(false);
+    agHelper.Sleep(2000);
+    agHelper.AddDsl("autoLayoutAutoHeight");
+  });
+
+  it("1. Should increase the height of the text widget when text is added", () => {
+    entityExplorer.ExpandCollapseEntity("Widgets");
+    agHelper.GetNClick(locators._widgetInCanvas(draggableWidgets.INPUT_V2));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoLayoutAutoHeight_spec.ts:24]: Please use -    ` EditorNavigation.SelectEntityByName("Button1", EntityType.Widget);`

**Comment by Aishwarya-U-R** [comment]
@riodeuno - Since here all other tests passed only [new test failed in last run](https://github.com/appsmithorg/appsmith/pull/29071#issuecomment-1827686398), please consider using /ci-test-limit for faster results on this spec fix only.

**Comment by riodeuno** [comment]
@Aishwarya-U-R Thanks for sharing this. I wasn't aware of this feature.

**Comment by Aishwarya-U-R** [comment]
Can I quick try it on this branch once? So you can see whats the file change needed?

**Comment by riodeuno** [comment]
@Aishwarya-U-R Yes, please feel free to do so

**Comment by Aishwarya-U-R** [comment]
@riodeuno - Now that limited-tests.txt is limited, you can run either /ci-test-limit or /ci-test-limit runId=xxxxx some docker built latest run id here that will be used to run only this spec.

**Review by marks0351 — APPROVED** [review]

**Review by prsidhu — APPROVED** [review]

**Pull request merged**